src/Enzyme.jl

```diff
@@ -1494,6 +1422,7 @@ end
     end
 end
 
+#WARN: this seems broken, doesn't have ReverseMode constructor
```

Comment:
how is this broken?

Reply:
`ReverseMode` lacks a 4th type parameter which it appears to need, or else there is no valid constructor and this gives a method error.  I attempted to fix in a new commit below by forwarding the parameter from the argument, though I did not look into whether this is appropriate.